src/main/java/com/google/api/generator/engine/ast/TypeNode.java

```diff
@@ -66,6 +67,14 @@
       withReference(ConcreteReference.withClazz(Short.class));
 
   private static final Map<TypeNode, TypeNode> BOXED_TYPE_MAP = createBoxedTypeMap();
+  private static final EnumSet<TypeKind> NUMERIC_TYPE_SET =
+      EnumSet.of(
```

Comment:
I see what you're trying to do here, and the intent is on the right track. However, boxing makes this more complicated because of the other object types, so consider using a set of `TypeNode`s instead and leveraging `equals()` to take care of that.

For background, `isPrimitive` is different because it distinguishes between object and non-object types.